### PR TITLE
feat: add reflector

### LINKED ISSUE #5: Mesh Reflector Material
**Is your feature request related to a problem? Please describe.**
Shining floors everywhere!!

**Describe the solution you'd like**
Same concept as R3F Drei https://github.com/pmndrs/drei#meshreflectormaterial 

**Suggested solution**
Shaders

### PR DESCRIPTION
**Pull request by JaimeTorrealba**

### CODE REVIEW

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Hi @JaimeTorrealba excellent job, I love the result.

I think we should add this to abstractions instead of materials because this implementation actually doesn't solve #5, **it's not a Material** but I see it as a handy abstraction from 'three/addons/objects/Reflector' 

The #5 is an actual material, to be used like this:

```
<TresMesh >
      <TresPlaneGeometry />
      <MeshReflectorMaterial blur="[300, 100]" resolution="2048" />
</TresMesh>
```

Another question would be, how are you handling the reactivity of those props?

File: docs/guide/materials/reflector.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			#include <logdepthbuf_fragment>
+
+			vec4 base = texture2DProj( tDiffuse, vUv );
+			gl_FragColor = vec4( blendOverlay( base.rgb, color ), 1.0 );
+
+			#include <tonemapping_fragment>
+			#include <colorspace_fragment>
+
+		}`
+```
```
Comment (alvarosabu) [inline docs/guide/materials/reflector.md:117]: When you add a new abstraction, try to be consistent with the documentation, the rest of the docs pages have a section called **Props** with a markdown table of the props. Please add it.

## Props

| Prop             | Description                                          | Default       |
| :--------------- | :--------------------------------------------------- | ------------- |
| `speed`          | Floating speed, higher it rocks more 🤘.             | `1`           |
| `rotationFactor` | Factor for Euler rotation.                           | `1`           |
| `floatFactor`    | Factor for Up/down movement.                         | `1`           |
| `range`          | Range of y-axis values the object will float within. | `[-0.1, 0.1]` |

**Review by alvarosabu — CHANGES_REQUESTED** [review]

**Comment by alvarosabu** [comment]
Docs build is broken, can you check the build locally?
 is broken.

File: docs/guide/materials/reflector.md (hunk 1 above)
Comment (JaimeTorrealba) [reply docs/guide/materials/reflector.md:117]: I agree i'll update the docs

**Comment by JaimeTorrealba** [comment]
@alvarosabu could be related to this doc's problem https://github.com/Tresjs/cientos/issues/137

**Comment by JaimeTorrealba** [comment]
In relation to the other comment, You're right is not a material, I'll move it, but It solves the exact same problem of the #5.
Are you sure you don't want to close it?? I mean obviously need some changes in the shaders but the effect you can achieve is the same

As the properties are in the args (And because of the way the Reflector class have been built in, I couldn't figure out how to make reactive props) only the color is reactive

**Comment by alvarosabu** [comment]
For the ticket leave it open, the idea would be to eventually create te reflector material.

For the reactivity I give you a small clue since you can get the reference of the instance, if one of the argument changes (watch) you could potentially dispose it and re initialize it with the new args. 

Give it a though 😉

**Comment by JaimeTorrealba** [comment]
I tried already but nothing work, nor dispose the material, geometry.

Nothing works, I Guess I am doing something wrong :(

**Comment by alvarosabu** [comment]
@JaimeTorrealba can you merge latest main and update this PR, it has been around for long

**Comment by JaimeTorrealba** [comment]
@alvarosabu I already merge main in this one.

Is not ready yet for merging to main, because is not reactive, when you have some time check it please. If you have any doubts, please let me know.

**Review by andretchen0 — CHANGES_REQUESTED** [review]

**Review by andretchen0 — CHANGES_REQUESTED** [review]
Looks good. Here are my suggestions.

File: docs/.vitepress/config.ts [hunk 2]
```diff
@@ -64,7 +65,6 @@ export default defineConfig({
         collapsed: true,
         items: [
           { text: 'WobbleMaterial', link: '/guide/materials/wobble-material' },
-          { text: 'MeshGlassMaterial', link: '/guide/materials/glass-material' },
```
[the remark this answers is not in the document]
Comment (JaimeTorrealba) [reply docs/.vitepress/config.ts]: I probably screw up at some points with the merge, thanks for letting me know this is very important

File: docs/guide/abstractions/reflector.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+## Usage
+
+```html{2}
+<Reflector :rotation="[-Math.PI * 0.5, 0, 0]" :position-y="-2" color="#F7F7F7">
+  <TresCircleGeometry :args="[10, 10]" />
+</Reflector>
+```
+
+| Prop              | Description                                          | Default                   |
```
[the remark this answers is not in the document]
Comment (JaimeTorrealba) [reply docs/guide/abstractions/reflector.md:14]: I remove the icons, and let just the warning message, the icons seems to be a little too much

File: docs/guide/abstractions/reflector.md [hunk 4]
```diff
@@ -0,0 +1,133 @@
+# Reflector
+
+The `cientos` package provides an abstraction of the Reflector class, which creates a reflector material in real time of your scene `<Reflector >` component that makes a geometry as an argument to create a mirror. This Mesh extends from `Mesh` so all the default props can be passed as well:
```
[the remark this answers is not in the document]
Comment (JaimeTorrealba) [reply docs/guide/abstractions/reflector.md]: Nice suggestion

**Review by andretchen0 — APPROVED** [review]
Looks good to me!

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Hey @JaimeTorrealba awesome to see this updated, can you add a demo to the documentation, this one is especially visual, it would be nice.

**Pull request merged**